backend/fms_core/services/experiment_run_info.py

```diff
@@ -0,0 +1,353 @@
+            if library.library_selection.name == 'Capture':
+                capture_details = _get_capture_details(derived_sample)
+                row.capture_kit = capture_details['capture_kit']
+                row.capture_baits = capture_details['capture_baits']
+            elif library.library_selection.name == "ChIP-Seq":
+                row.chip_seq_mark = library.library_selection.target
+
+    return row
+
+def _convert_flowcell_coordinate_to_lane_number(coord: str, container: Container) -> int:
+    '''Converts a freezeman coordinate (eg. A01) to an integer flowcell lane number.'''
+
+    container_spec = CONTAINER_KIND_SPECS[container.kind]
+    if container_spec is None:
+        raise Exception(f'Cannot convert coord {coord} to lane number. No ContainerSpec found for container kind "{container.kind}".')
+
+    lane = convert_alpha_digit_coord_to_ordinal(coord, container_spec.coordinate_spec)
+    return lane
+
+def _find_library_prep(library: Library) -> Union[ProcessMeasurement, None]:
+    ''' 
+    Given a library, find the Library Preparation process that was used to create the
+    library, if any, to extract the library preparation properties (eg. Library Kit).
+
+    Note that not all libraries will have a Library Preparation measurement,
+    since they can be imported directly into freezeman without passing the
+    Library Preparation step.
+    '''
+    
+    library_prep: Union[ProcessMeasurement, None] = None
+    try:
+        library_to_find : Library = library
+
+        if library.library_selection is not None:
+            # If a library was captured then that generated a new library instance.
+            # We have to find the previous instance, on which the Library Prep was run.
+            try:
+                capture_protocol = Protocol.objects.get(name="Library Capture")
+                capture_lineage: SampleLineage = SampleLineage.objects.get(
+                    process_measurement__process__protocol_id=capture_protocol.pk,
+                    child__derived_samples__library__id=library.pk
+                )
+                #library_to_find = Library.objects.get(derived_sample__biosample_id=derived_sample.biosample.id)
+                # Now find the derived sample in the parent that contains the same biosample id as our library.
+                library_to_find = Library.objects.get(derived_sample__samples=capture_lineage.parent, derived_sample__biosample=library.derived_sample.biosample)
+                
+            except:
+                pass
+
+        # Find the library preparation process measurement
+        protocol: Protocol = Protocol.objects.get(name="Library Preparation")
+        lineage: SampleLineage = SampleLineage.objects.get(
+            process_measurement__process__protocol_id=protocol.pk,
+            child__derived_samples__library__id=library_to_find.pk)
+        library_prep = lineage.process_measurement
+    except SampleLineage.DoesNotExist:
```

Comment:
do not shadow other "try...except" unless it is needed. Here the only risky calls are the lineage get() which should be dealt with individually and separately (at the same level, not one within the other.)